### ISSUE TITLE
accessing azure blob storage using managed identity

### ISSUE THREAD

**Issue by neelam256** [Azure/azure-storage-cpp#338]
I want to access azure blob storage from container, and Container is running under managed identity. Can you please point out some cpp example to access blob storage.

**Comment by katmsft** [comment]
This SDK currently supports authorizing with OAuth Bearer token, however, there is currently no support on acquiring such token via the SDK. The token should be acquired with a designated Azure Identity library, which unfortunately lacks CPP implementation at the moment.

You can reference the below sample to see how to use the token:
https://github.com/Azure/azure-storage-cpp/blob/master/Microsoft.WindowsAzure.Storage/samples/OAuthGettingStarted.cpp

**Comment by neelam256** [comment]
@katmsft thanks for response, is it intended to support the acquisition of token via SDK in the future ?

**Comment by katmsft** [comment]
Yes, Identity SDK is on our Roadmap but unfortunately, that would come with a newer version of the Storage SDK that does not have a similar API signature and dependency list comparing with this one. The new SDK is months away but should have a good shape by the end of the year.

**Comment by neelam256** [comment]
@katmsft  thank you update, Is there anyway i can track this ?
I have another question for accessing storage account,
I used example , two cases 
first: I generate access token for mystorage account  with 
"http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&amp;resource=https://<mystorageaccount>.blob.core.windows.net" 
I can access storage account.
But when I use Access token generated by "http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https://management.azure.com/" 
I got an error
Error::Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.

As this storage account belongs to this Azure resource group, It should be accessible.

**Comment by Jinming-Hu** [comment]
@nilact This sdk has been deprecated in favor of our [next generation of storage SDK](https://github.com/Azure/azure-sdk-for-cpp), which supports managed identity.
